### ISSUE TITLE
int64 format not specified in some (at least one?) location

### ISSUE THREAD

**Issue by philipbawn** [openzipkin/zipkin-api#93]
https://github.com/openzipkin/zipkin-api/blob/master/zipkin2-api.yaml line 319 has type: integer, but format: int64 is not specified. due to this, my client generator (nswag) made the Annotation.timestamp property an int32 which of course blows up. Is this yml file the source of truth or is something generating it? If the yaml file can just be edited i can go through it and see if there is this problem anywhere else and make a PR.

Thanks.

**Comment by devinsba** [comment]
The YAML file is maintained by hand. A contribution would be very welcome, thanks

**Comment by jcchavezs** [comment]
+1

José Carlos Chávez


ons. 24. feb. 2021 kl. 00:44 skrev Brian Devins-Suresh <
notifications@github.com>:

**Comment by philipbawn** [comment]
I'm looking through it now as well as the output from my client generator :)

**Comment by philipbawn** [comment]
This is solved by this PR https://github.com/openzipkin/zipkin-api/pull/94 which makes the swagger file work well with nswag. I think write operations were already OK.

**Comment by philipbawn** [comment]
You know what.. I don't see how writes would work if someone generated a client using this before the modification. Here is what the current master [7692ca7](https://github.com/openzipkin/zipkin-api/commit/7692ca7be4dc3be9225db550d60c4d30e6e9ec59) looks like on my screen:

![span](https://user-images.githubusercontent.com/10352261/109091459-af827a80-76c9-11eb-9286-729f0e26ebd7.PNG)

Maybe I'm agonizing over this too much but I do worry about breaking other people's stuff.

Here is my client after the fix from the PR.
![span2](https://user-images.githubusercontent.com/10352261/109091690-1a33b600-76ca-11eb-8bec-7df938519a65.PNG)

That looks really good. I'm using NSwag v13.0.5.0

**Comment by philipbawn** [comment]
The opentelemetry-dotnet people appear to have wrote their [own class](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Exporter.Zipkin/Implementation/ZipkinSpan.cs) (rather than generate one from the swagger file) which makes use of long data type, similar to my result after adding the format to this YAML file. This must be why I did not encounter any problems sending telemetry to zipkin but only querying from it when I raised the issue. I am using their package for sending telemetry.
